packages/components/tests/dummy/app/templates/components/code-block.hbs

```diff
@@ -77,6 +79,34 @@ console.log(`I am ${codeLang} code`);"
     </SG.Item>
   </Shw::Grid>
 
+  <Shw::Divider @level="2" />
+
+  <Shw::Text::H3>Dynamic content</Shw::Text::H3>
+
+  <Shw::Text::Body>
+    <button type="button" {{on "click" this.updateCodeValue}}>
+      Update value
+    </button>
+  </Shw::Text::Body>
+
+  <Shw::Text::Body>
+    <label for="input">Change input value:</label>
+    <input id="input" type="text" value={{this.input}} {{on "input" this.updateInput}} />
+  </Shw::Text::Body>
+
+  <Hds::CodeBlock
+    @value={{this.codeValue}}
+    @language="javascript"
+    @hasCopyButton={{true}}
+    @hasLineNumbers={{false}}
```

Comment:
I noticed that if you set this to "true" (and probably we should) when the lines gets updated, the line numbers don't.

Reply:
I looked into this during the first week of January and today again. Unfortunately, I can not see a way in which the `@hasLineNumbers` feature would work with dynamic content.

For this reason, I suggest the pragmatic approach of fixing the immediate issues (I updated the PR description to highlight what this PR covers and what doesn't) to address current instances in HCP and Boundary then leave the `hasLineNumbers` issue for future evaluation (it may require us to fork the addon or move away from Prism altogether). If you find this approach acceptable, I can create a ticket to track this use case and update the CodeBlock guidance to call out the current limitations.